servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/NettyPipelinedConnection.java

```diff
@@ -0,0 +1,353 @@
+                          final Supplier<FlushStrategy> flushStrategySupplier,
+                          final Supplier<WriteDemandEstimator> writeDemandEstimatorSupplier) {
+        return Publisher.defer(() -> {
+            // Lazy modification of local state required (e.g. nodes, delayed subscriber, queue modifications)
+
+            // Setup read side publisher and nodes
+            DelayedSubscribePublisher<Resp> delayedReadPublisher = new DelayedSubscribePublisher<>(
+                    toSource(connection.read()));
+            Node readNode = new Node() {
+                @Override
+                public void run() {
+                    try {
+                        delayedReadPublisher.processSubscribers();
+                    } catch (Throwable cause) {
+                        closeOnError(connection, delayedReadPublisher, cause);
+                    }
+                }
+            };
+            Publisher<Resp> composedReadPublisher = delayedReadPublisher.liftSync(new ReadPopNextOperator(readNode));
+
+            // Setup write side publisher and nodes
+            DelayedSubscribeCompletable delayedWriteCompletable = new DelayedSubscribeCompletable(toSource(
+                            connection.write(requestPublisher, flushStrategySupplier, writeDemandEstimatorSupplier)));
+            Node writeNode = new Node() {
+                @Override
+                public void run() {
+                    try {
+                        try {
+                            readQueue.offer(readNode);
+                        } catch (Throwable cause) {
+                            delayedWriteCompletable.failSubscribers(cause);
+                            // We have not started the write at this point, and the above operation will:
+                            // - fail the write subscriber (pop the next write). We haven't subscribed to
+                            // Completable returned from connection.write(..) and therefore the connection hasn't
+                            // subscribed to the requestPublisher so there is nothing to cancel on the write side.
+                            // - fail the merge operator (pop the next read, if still in progress), cancel the
+                            // connection.read() (which may close the connection if the read is still active), and
+                            // deliver an error to the downstream read Subscriber.
+                            return;
+                        }
+
+                        delayedWriteCompletable.processSubscribers();
+                    } catch (Throwable cause) {
+                        closeOnError(connection, delayedWriteCompletable, cause);
+                    }
+                }
+            };
+
+            try {
+                writeQueue.offer(writeNode);
+            } catch (Throwable cause) {
+                // The writeNode's run method handles exceptions as a result of Subscribe/external calls. An Exception
```

Comment:
@NiteshKant - this exception handling was added for demonstration purposes but is somewhat superfluous bcz the queue offer doesn't do any allocations (e.g. node allocation done outside the scope), and we don't always exhaustively protect against OOME type errors.

also the try/catch around the insertions are also somewhat much, because the `Publisher`  (or completable) will catch exceptions thrown during `subscribe` (in the internal `handleSubscribe` method). Let me know if you think it is worth keeping all of these cases.